src/sync/branches.ts

```diff
@@ -6,25 +6,28 @@ import { BackfillMessagePayload } from "~/src/sqs/sqs.types";
 
 // TODO: better typings
 export const getBranchTask = async (
-	logger: Logger,
+	parentLogger: Logger,
 	gitHubClient: GitHubInstallationClient,
 	_jiraHost: string,
 	repository: Repository,
 	cursor: string | undefined,
 	perPage: number,
 	messagePayload: BackfillMessagePayload) => {
 
-	logger.info("Syncing branches: started");
+	const logger = parentLogger.child({ backfillTask: "Branch" });
+	const startTime = Date.now();
+
+	logger.info({ startTime }, "Backfill task started");
 
 	const commitSince = messagePayload.commitsFromDate ? new Date(messagePayload.commitsFromDate) : undefined;
 	const result = await gitHubClient.getBranchesPage(repository.owner.login, repository.name, perPage, commitSince, cursor as string);
 	const edges = result?.repository?.refs?.edges || [];
 	const branches = edges.map(edge => edge?.node);
 
-	logger.debug("Syncing branches: finished");
-
 	const jiraPayload = transformBranches({ branches, repository }, messagePayload.gitHubAppConfig?.gitHubBaseUrl);
 
+	logger.info({ processingTime: Date.now() - startTime, jiraPayloadLength: jiraPayload?.branches?.length }, "Backfill task complete");
```

Comment:
Again, that `{ processingTime: Date.now() - startTime` should be somewhere upper level in sync/installation.ts
However I understand why here, as we need some more logging for `jiraPayload?.branches?.length`. 

I want to understand the purpose of the changes here. If it is for auditing problem, we might need to put more info in it. If it is for performance reason, we probably should use statsD instead.

Reply:
You are correct! the step after this is to emit that processingTime with StatsD, whether it needs to be in the log or not.... maybe/maybe not but it was floating around :lol: , this log was mainly just to see where it returns, so we can trace the flow if we need to.

Same as before, its an existing log, just aligning them across all the tasks